api/src/main/java/com/capitalone/dashboard/service/CollectorServiceImpl.java

```diff
@@ -67,8 +67,12 @@ public CollectorServiceImpl(CollectorRepository collectorRepository,
                 jobName =  findIndex(descriptionFilter);
             }
         }
-        if(!niceName.isEmpty() && collectorType == CollectorType.Build){
-           collectorItems = collectorItemRepository.findByCollectorIdInAndDescriptionContainingAndNiceNameContainingAllIgnoreCase(collectorIds, jobName,niceName, pageable);
+        if(collectorType == CollectorType.Build){
+            if(!niceName.isEmpty()){
```

Comment:
We need to have Collector specific implementations in here. If we are looking to target a field to search on, should we create a new json element called "searchField" or something in the collector items and have it populated by collector (which is tool/type specific)?